### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by Joozty**

### CODE REVIEW

**Comment by viktor-yakubiv** [comment]
Can we fallback to simple _Logout_ link without anything else? It might even better if it stays pure link to `API/logout`

**Comment by Joozty** [comment]
@viktor-yakubiv please have a look now.

File: store/root.js [hunk 1]
```diff
@@ -47,6 +47,13 @@ class Root {
     }
   }
 
+  // eslint-disable-next-line class-methods-use-this
+  async logout() {
+    // TODO: /logout should ideally be moved to /internal
+    await apiRequest('https://api.dev.core.ac.uk/logout')
+    window.location.replace('/logout.html')
+  }
```
Comment (viktor-yakubiv) [inline store/root.js]: Do you add this method just because to do not like having link to API in the UI?

File: components/layout/app-bar.jsx [hunk 2]
```diff
@@ -25,8 +26,14 @@ const DashboardAppBar = ({ children, className, ...restProps }) => {
       </AppBar.Brand>
 
       {children}
+      <AppBar.Item>
+        {`${store.user.givenName} ${store.user.familyName} `}|{' '}
+        <Button className={styles.logoutButton} onClick={store.logout}>
+          Logout
+        </Button>
+      </AppBar.Item>
```
Comment (viktor-yakubiv) [inline components/layout/app-bar.jsx]: Can you write this as a separate component that will be connected to store but the AppBar will not?

File: components/layout/app-bar.jsx [hunk 3]
```diff
@@ -25,8 +26,14 @@ const DashboardAppBar = ({ children, className, ...restProps }) => {
       </AppBar.Brand>
 
       {children}
+      <AppBar.Item>
+        {`${store.user.givenName} ${store.user.familyName} `}|{' '}
```
Comment (viktor-yakubiv) [inline components/layout/app-bar.jsx]: Could you pleas add here some `span` with class that would hide on mobile screens?

File: components/layout/app-bar.jsx [hunk 4]
```diff
@@ -25,8 +26,14 @@ const DashboardAppBar = ({ children, className, ...restProps }) => {
...
+        <Button className={styles.logoutButton} onClick={store.logout}>
```
Comment (viktor-yakubiv) [inline components/layout/app-bar.jsx]: I would prefer this being pure link.

**Comment by Joozty** [comment]
@viktor-yakubiv That's why I created dropdown menu.

File: components/layout/app-bar.jsx (hunk 4 above)
Comment (Joozty) [reply components/layout/app-bar.jsx]: How? It cannot be a pure link because the cookie won't be deleted.

File: store/root.js (hunk 1 above)
Comment (Joozty) [reply store/root.js]: I don't mind having API in UI. But in that case, logout has to work like this `https://api.dev.core.ac.uk/logout?continue=someurl`. 

Also, it's nice to have some action in the store. For example, when we store something in local storage this is the right place for removing it from there.

File: components/layout/app-bar.jsx (hunk 4 above)
Comment (viktor-yakubiv) [reply components/layout/app-bar.jsx]: Link to `api.core.ac.uk/logout?continue=/logout.html`

File: components/layout/app-bar.jsx (hunk 2 above)
Comment (Joozty) [reply components/layout/app-bar.jsx]: sure

File: store/root.js (hunk 1 above)
Comment (viktor-yakubiv) [reply store/root.js]: I agree. On the other hand, having another action that can be avoided is not that good. Less code is less errors what it better.

**Comment by Joozty** [comment]
@viktor-yakubiv I did changes you requested. As you said we should distinguish API URL based on NODE_ENV. I leave it as is for now. It should be handled in a separate PR. 

NOTE: It doesn't work now. I think @valerapro will deploy that continue param tomorrow.

File: components/application/logout.jsx [hunk 5]
```diff
@@ -7,7 +7,7 @@ import { withGlobalStore } from 'store'
 
 const LOGOUT_URL = `https://api.dev.core.ac.uk/logout?continue=${encodeURI(
   typeof window !== 'undefined'
-    ? `${window.location.protocol}//${window.location.host}`
+    ? `${window.location.protocol}//${window.location.host}/logout.html`
```
Comment (viktor-yakubiv) [inline components/application/logout.jsx]: You may find useful [`window.location.origin`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/origin)

File: components/application/logout.jsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ import { withGlobalStore } from 'store'
 
 const LOGOUT_URL = `https://api.dev.core.ac.uk/logout?continue=${encodeURI(
```
Comment (viktor-yakubiv) [inline components/application/logout.jsx]: Perhaps, you meant [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)

File: components/application/logout.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+const LOGOUT_URL = `https://api.dev.core.ac.uk/logout?continue=${encodeURI(
+  typeof window !== 'undefined'
+    ? `${window.location.protocol}//${window.location.host}/logout.html`
+    : ''
+)}`
+
+const Logout = ({ store }) => (
+  <AppBar.Item className={styles.container}>
+    {`${store.user.givenName} ${store.user.familyName} `}|{' '}
```
Comment (viktor-yakubiv) [inline components/application/logout.jsx]: Can this be `<span className={styles.name}>{name}</span>`?

where 

```js
name = `${store.user.givenName} ${store.user.familyName}` || store.user.email`
```

and

```css
.name {
  &::after { content: ' | ' }
  @media (--screen-until-desktop) { display: none }
}
```

**Pull request merged**